util-cache/src/main/scala/sbt/util/FileInfo.scala

```diff
@@ -4,13 +4,18 @@
 package sbt.util
 
 import java.io.File
+
 import scala.util.control.NonFatal
 import sbt.io.{ Hash, IO }
-import sjsonnew.{ Builder, JsonFormat, Unbuilder, deserializationError }
-import CacheImplicits._
+import sjsonnew.{ Builder, DeserializationException, JsonFormat, Unbuilder, deserializationError }
+import CacheImplicits.{ arrayFormat => _, _ }
 
 sealed trait FileInfo { def file: File }
-sealed trait HashFileInfo extends FileInfo { def hash: List[Byte] }
+sealed trait HashFileInfo extends FileInfo {
+  @deprecated("Use hashArray instead", "1.3.0")
+  def hash: List[Byte] = hashArray.toList
+  private[util] def hashArray: Array[Byte]
```

Comment:
Should this be public if you want people migrate to hashArray?

Reply:
I made it private to ensure immutability. I'm not sure whether these traits should be public or not, I don't see any reason for them to be used directly.

Reply:
`FileInfo.hash` is public, and it returns `HashFileInfo`. I guess there's no reason peope would be called `hash` directly, so it should be ok.